core/modules/user/user.install

```diff
@@ -1138,5 +1137,20 @@ function user_update_1012() {
 }
 
 /**
+ * Grant accounts with 'administer permissions' the new 'assign roles' permission.
+ */
+function user_update_1013() {
+  $names = config_get_names_with_prefix('user.role.');
+  foreach ($names as $config_name) {
+    $config = config($config_name);
+    $permissions = $config->get('permissions');
+    if (in_array('administer permissions', $permissions)) {
+      $permissions[] = 'assign roles';
+      $config->save();
```

Comment:
This Update hook doesn't seem to work. It's missing the assignment of $permissions back into $config->set('permissions', $permissions).